pkg/sparrow/targets/remote/gitlab/gitlab.go

```diff
@@ -56,184 +53,202 @@ type Config struct {
 	Token string `yaml:"token" mapstructure:"token"`
 	// ProjectID is the ID of the project in the gitlab instance that contains the global targets
 	ProjectID int `yaml:"projectId" mapstructure:"projectId"`
+	// Branch is the branch to use for the gitlab repository
+	Branch string `yaml:"branch" mapstructure:"branch"`
 }
 
 func New(cfg Config) remote.Interactor {
-	return &client{
-		baseUrl:   cfg.BaseURL,
-		token:     cfg.Token,
-		projectID: cfg.ProjectID,
-		client:    &http.Client{},
+	c := &client{
+		config: cfg,
+		client: &http.Client{
+			Timeout: 30 * time.Second,
+		},
 	}
+	if c.config.Branch == "" {
+		c.config.Branch = c.fetchDefaultBranch()
+	}
+	return c
 }
 
 // FetchFiles fetches the files from the global targets repository from the configured gitlab repository
 func (c *client) FetchFiles(ctx context.Context) ([]checks.GlobalTarget, error) {
 	log := logger.FromContext(ctx)
 	fl, err := c.fetchFileList(ctx)
 	if err != nil {
-		log.Error("Failed to fetch files", "error", err)
+		log.ErrorContext(ctx, "Failed to fetch files", "error", err)
 		return nil, err
 	}
 
 	var result []checks.GlobalTarget
 	for _, f := range fl {
 		gl, err := c.fetchFile(ctx, f)
 		if err != nil {
-			log.Error("Failed fetching files", "error", err)
+			log.ErrorContext(ctx, "Failed fetching files", "error", err)
 			return nil, err
 		}
 		result = append(result, gl)
 	}
-	log.Info("Successfully fetched all target files", "files", len(result))
+	log.InfoContext(ctx, "Successfully fetched all target files", "files", len(result))
 	return result, nil
 }
 
 // fetchFile fetches the file from the global targets repository from the configured gitlab repository
-func (c *client) fetchFile(ctx context.Context, f string) (checks.GlobalTarget, error) {
+func (c *client) fetchFile(ctx context.Context, f string) (res checks.GlobalTarget, err error) {
```

Comment:
I only add named returns on one of the following conditions:

1. If it improves readability, especially when returning the same type multiple times or more than three return values.
2. If these variables are used in a `defer` function to further change the returned values. This is AFAIK the only option to return an error from a deferred function.

In this case the second condition is true because I use the `err` variable to join the body closing error with the other error return value. If we decide to ignore the response body closing errors, this variable would be obsolete again and should be removed.

Reply:
Ok after delving more into the [Go GC](https://tip.golang.org/doc/gc-guide#Where_Go_Values_Live) and the [go language spec](https://tip.golang.org/ref/spec#Declarations_and_scope), I've come to the conclusion that my second condition is not entirely true. This is because inner function variable scopes either end in the innermost containing block for stack allocations (here the deferred function), or if the underlying value is allocated on the heap after the GC determined that there are no remaining references to it.

This means I'll remove all of the named return values and use normal variables instead. Thanks for commenting this, otherwise I'd probably not have read the blog post 😄

Reply:
@puffitos I think we made a mistake back then by removing the named returns. I've just tested this in a [go playground](https://go.dev/play/p/WAlSbpL6XG-) and it seems that the deferred function can't manipulate the return values if they are not named.

This means that since this change all of our deferred functions don't work as expected and we may have ignored some errors since then. 

To still address your concern about named returns not being readable enough, I'd propose the following solution:

```go
func foo() error {
    _, err := bar()
    return err
}

// Note that we only use the named return value for the error in order to be able to manipulate it in the deferred function
// The other return value is not named since it's not used in the deferred function
func bar() (_ int, err error) {
    defer func(){
        err = errors.Join(err, errors.New("additional error"))
    }
    return 0, errors.New("error")
}
```

This way we can still manipulate the error in the deferred function and don't have to use named return values for all return values.

What do you think about this solution?

Reply:
The http response body is typically an `http2transportResponseBody`, whose `Close` only returns an error if the request gets cancelled (we're not cancelling any requests AFAIK):

https://cs.opensource.google/go/go/+/master:src/net/http/h2_bundle.go;l=9822?q=http2transport&ss=go%2Fgo

Feel free to check the http package for other implementations of the Close() error function:

https://cs.opensource.google/search?q=%22Close()%20error%22&ss=go%2Fgo:src%2Fnet%2Fhttp%2F

My point here is, that there's a reason why the go documentation just ignores the error:
https://pkg.go.dev/net/http#pkg-overview

It typically doesn't seem to happen, and even if it does, we should be able to easily spot the memory leak. Let's stop worrying about how we join errors. Not every edge case is worth following, if this means that we add named returns in multiple functions and defer error handling. That's all just too much effort, which adds complexity to our codebase for a return on investment that's neither measurable nor visible.

Reply:
Ok I see your point, after trying to find the correct implementation of the `Close` method. It's rather complex to find it since it depends on many factors like the underlying transport, the connection state, etc.

Let's gradually remove all response body error handling and be aware of possible memory leaks. If we find any, we can still add the error handling back in. Thanks for the discussion, I think we've found a good solution here.